octree/basic.go

```diff
@@ -131,8 +92,25 @@ func (octree *basicOctree) At(x, y, z float64) (pc.Data, bool) {
 	return nil, false
 }
 
-// Iterate TODO: Implement iterate for octree.
-func (octree *basicOctree) Iterate(numBatches, myBatch int, fn func(p r3.Vector, d pc.Data) bool) {
+// Iterate is a batchable process that will go through a basic octree and apply a specified function
+// to either all the data points or a subset of them based on the given numBatches and currentBatch
+// inputs. If any of the applied functions returns a false value, iteration will stop and no further
+// points will be processed.
+func (octree *basicOctree) Iterate(numBatches, currentBatch int, fn func(p r3.Vector, d pc.Data) bool) {
```

Comment:
That is not a requirement any of the other iterate function. In addition if currentBatch > numBatches the upper and lower bounds will be outside the array and trigger (lowerBound < upperBound) to return false, thus not performing any of the iterate. 

Since Iterate does not return an error this behavior should be acceptable